### ISSUE TITLE
Unable to Reproducing Tox21 Results

### ISSUE THREAD

**Issue by mu-wang** [deepchem/deepchem#1354]
I have difficulty reproducing the Tox21 results shown on MoleculeNet.ai.

I ran the following code 3 times, The `roc_auc_score` of the validation scores are `[0.814, 0.807, 0.826]`, consistently lower than the benchmark.
```python
"""
Script that trains graph-conv models on Tox21 dataset.
"""
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import numpy as np

# np.random.seed(123)
import tensorflow as tf

# tf.set_random_seed(123)
import deepchem as dc
from deepchem.molnet import load_tox21
from deepchem.models.tensorgraph.models.graph_models import GraphConvModel

model_dir = "/tmp/graph_conv"

# Load Tox21 dataset
tox21_tasks, tox21_datasets, transformers = load_tox21(featurizer='GraphConv')
train_dataset, valid_dataset, test_dataset = tox21_datasets
print(train_dataset.data_dir)
print(valid_dataset.data_dir)

# Fit models
metric = dc.metrics.Metric(
    dc.metrics.roc_auc_score, np.mean, mode="classification")

# Batch size of models
batch_size = 16

model = GraphConvModel(len(tox21_tasks), 
                       graph_conv_layers=[64, 128, 64], 
                       dense_layer_size=128, 
                       dropout=[0.1, 0.1, 0.1, 0.3],
                       batch_size=batch_size, mode='classification')

model.fit(train_dataset, nb_epoch=50)

print("Evaluating model")
train_scores = model.evaluate(train_dataset, [metric], transformers)
valid_scores = model.evaluate(valid_dataset, [metric], transformers)

print("Train scores")
print(train_scores)

print("Validation scores")
print(valid_scores)
```

Not sure if there are other parameter settings that I have not correctly capture for the training process? I am using deepchem 2.1.0.

**Comment by rbharath** [comment]
The MoleculeNet hyperparameters were set by Gaussian process search. I think you need to re-use those exact hyperparameters to reproduce results exactly.

@miaecle We have a copy of the best hyperparameters stored somewhere right?

**Comment by miaecle** [comment]
@mu-wang @rbharath Yes they are stored here:
http://deepchem.io.s3-website-us-west-1.amazonaws.com/trained_models/Hyperparameter_MoleculeNetv3.tar.gz
Actually the current graphconv model might not be the exact implementation used in the benchmark though. Other than that, random split is applied in the benchmark run, also I don't recall that we applied dropout in the models as well.

**Comment by rbharath** [comment]
Ah good point. The current graph-conv has been refactored to increase training speed, which might potentially effect the choice of hyperparameters. For best results, I'd recommend running a new hyperparameter search to find best values.

**Comment by mu-wang** [comment]
OK. I will try a new Gaussian hyperparameter search. @rbharath @miaecle Thank you both for your help!

**Comment by rbharath** [comment]
I'm going to close this old issue. Please re-open if you still have questions!
